miio/gateway.py

```diff
@@ -126,58 +122,49 @@ def light(self) -> "GatewayLight":
         """Return light control interface."""
         return self._light
 
-    @command()
+    @property
     def devices(self):
-        """Return list of devices."""
-        # from https://github.com/aholstenson/miio/issues/26
-        devices_raw = self.send("get_device_prop", ["lumi.0", "device_list"])
-        devices = [
-            SubDevice(self, *devices_raw[x : x + 5])  # noqa: E203
-            for x in range(0, len(devices_raw), 5)
-        ]
+        """Return a list of the already discovered devices."""
+        return self._devices
 
-        return devices
-
-    @command(click.argument("sid"), click.argument("property"))
-    def get_device_prop(self, sid, property):
+    @command()
+    def discover_devices(self):
+        """Discovers SubDevices and returns a list of the discovered devices."""
+        # from https://github.com/aholstenson/miio/issues/26
+        device_type_mapping = {"AqaraRelayTwoChannels": AqaraRelayTwoChannels, "Plug": Plug, "SensorHT": SensorHT, "AqaraHT": AqaraHT, "AqaraMagnet": AqaraMagnet}
+        devices_raw = self.get_gateway_prop("device_list")
+        self._devices = []
+        
+        for x in range(0, len(devices_raw), 5):
+            try:
+                Device_name = DeviceType(devices_raw[x+1]).name
+            except ValueError:
+                _LOGGER.warning("Unknown subdevice type '%i': %s discovered, of Xiaomi gateway with ip: %s", devices_raw[x+1], devices_raw[x], self.ip)
+                Device_name = DeviceType(-1).name
+            
+            SubDeviceClass = device_type_mapping.get(Device_name, SubDevice)
```

Comment:
How about simply checking for Noneness?
```suggestion
            subdevice_cls = device_type_mapping.get(Device_name)
```

Reply:
That is indeed better